components/camel-rabbitmq/src/main/java/org/apache/camel/component/rabbitmq/RabbitMQMessageConverter.java

```diff
@@ -217,4 +222,63 @@ private Object getValidRabbitMQHeaderValue(Object headerValue) {
 
         return null;
     }
+
+    boolean hasSerializeHeader(AMQP.BasicProperties properties) {
+        if (properties == null || properties.getHeaders() == null) {
+            return false;
+        }
+        if (properties.getHeaders().containsKey(RabbitMQEndpoint.SERIALIZE_HEADER) && properties.getHeaders().get(RabbitMQEndpoint.SERIALIZE_HEADER).equals(true)) {
+            return true;
+        }
+        return false;
+    }
+
+    public void populateRabbitExchange(Exchange camelExchange, Envelope envelope, AMQP.BasicProperties properties, byte[] body) {
+        Message message;
+        if (camelExchange.getIn() != null) {
+            // Use the existing message so we keep the headers
+            message = camelExchange.getIn();
+        } else {
+            message = new DefaultMessage();
+            camelExchange.setIn(message);
+        }
+
+        if (envelope != null) {
+            message.setHeader(RabbitMQConstants.ROUTING_KEY, envelope.getRoutingKey());
+            message.setHeader(RabbitMQConstants.EXCHANGE_NAME, envelope.getExchange());
+            message.setHeader(RabbitMQConstants.DELIVERY_TAG, envelope.getDeliveryTag());
+        }
+
+        Map<String, Object> headers = properties.getHeaders();
+        if (headers != null) {
+            for (Map.Entry<String, Object> entry : headers.entrySet()) {
+                // Convert LongStrings to String.
+                if (entry.getValue() instanceof LongString) {
+                    message.setHeader(entry.getKey(), entry.getValue().toString());
+                } else {
+                    message.setHeader(entry.getKey(), entry.getValue());
+                }
+            }
+        }
+
+        if (hasSerializeHeader(properties)) {
+            Object messageBody = null;
+            try (InputStream b = new ByteArrayInputStream(body);
+                 ObjectInputStream o = new ObjectInputStream(b);) {
+                messageBody = o.readObject();
+            } catch (IOException | ClassNotFoundException e) {
+                LOG.warn("Could not deserialize the object");
```

Comment:
We should not swallow exceptions. Maybe better to set the exception on the exchange or something ?